apm-sniffer/apm-agent/src/main/java/org/apache/skywalking/apm/agent/SkyWalkingAgent.java

```diff
@@ -161,9 +175,10 @@ public DynamicType.Builder<?> transform(final DynamicType.Builder<?> builder,
                                                 final JavaModule javaModule,
                                                 final ProtectionDomain protectionDomain) {
             LoadedLibraryCollector.registerURLClassLoader(classLoader);
+            DelegateNamingResolver.reset();
```

Comment:
I am a little confused about this `reset`. `#transform` should be called for every type, as a static method, we reset every time?

Reply:
If byte-buddy could expose the whole method signature rather than the name, I believe this would not be an issue.
How about asking for upstream first?

Reply:
What do you mean? Tell me more about it.